### ISSUE TITLE
@ binding reformatted strangely on rustfmt from rustc 1.41.0-beta.4

### ISSUE THREAD

**Issue by wezm** [rust-lang/rustfmt#4031]
The following line is formatting differently and in an undesirable way with rustfmt from rustc beta (1.41.0-beta.4):

https://github.com/yeslogic/allsorts/blob/c78fb470ca2a575570543918cfbf15422b0377b2/tests/woff2.rs#L236

The diff looks like this:

```diff
Diff in /home/travis/build/yeslogic/allsorts/tests/woff2.rs at line 233:
             .iter()
             .map(|glyph| match glyph {
                 GlyfRecord::Parsed(
-                    found @ Glyph {
+                    found
+                    @
+                    Glyph {
                         data: GlyphData::Composite { .. },
                         ..
                     },
```

**Comment by Follpvosten** [comment]
I still have this issue with a nested pattern on 1.57:
```rust
if let Some(
    mut
    ppart
    @
    PathPart {
        kind: PartKind::End { grid_width },
        ..
    },
) = ppart
{
    // ...
}
```
should I open a new bug for it?

**Comment by Follpvosten** [comment]
@calebcartwright Yes, I'm using stable, and good to know. If there is a fix that's about to be published in 1.59, that's good enough for me.
